src/lptim.rs

```diff
@@ -105,6 +111,81 @@ impl LpTimer<OneShot> {
     }
 }
 
+impl LpTimer<Encoder> {
+    /// Initializes the Low-Power Timer in encoder mode.
+    ///
+    /// The `start` method must be called to enable the encoder input.
+    pub fn init_encoder(
+        lptim: LPTIM,
+        pwr: &mut PWR,
+        rcc: &mut Rcc,
+        clk: ClockSrc,
+        (pb5, pb7): (gpiob::PB5<gpio::Analog>, gpiob::PB7<gpio::Analog>),
+    ) -> Self {
+        pb5.set_alt_mode(gpio::AltMode::AF2);
+        pb7.set_alt_mode(gpio::AltMode::AF2);
+
+        Self::init(lptim, pwr, rcc, clk)
+    }
+
+    // TODO: Dedupe this fn with configure() function in `impl<M: CountMode> LpTimer<M>`
+    fn configure_encoder(&mut self, arr: u16) {
+        // Disable the timer. The prescaler can only be changed while it's disabled.
+        self.lptim.cr.write(|w| w.enable().clear_bit());
+
+        // Configure in encoder mode
+        self.lptim.cfgr.write(|w| {
+            w
+                // Make sure prescaler is disabled. Encoder mode forbids prescaling.
+                .presc()
+                .div1()
+                // Put timer into encoder mode
+                .enc()
+                .set_bit()
+                // Choose internal clock source - external sources not supported in encoder mode.
+                .cksel()
+                .clear_bit()
+                // Start counting from software trigger
+                .trigen()
+                .sw()
+                // Clock polarity
+                .ckpol()
+                .both_edges()
+        });
+
+        // Enable timer
+        self.lptim.cr.write(|w| w.enable().set_bit());
+
+        // "After setting the ENABLE bit, a delay of two counter clock is needed before the LPTIM is
+        // actually enabled."
+        // The slowest LPTIM clock source is LSE at 32768 Hz, the fastest CPU clock is ~80 MHz. At
+        // these conditions, one cycle of the LPTIM clock takes 2500 CPU cycles, so sleep for 5000.
+        cortex_m::asm::delay(5000);
+
+        // ARR can only be changed while the timer is *en*abled
+        self.lptim.arr.write(|w| w.arr().bits(arr));
+    }
+
+    pub fn enable(&mut self, arr: u16) {
```

Comment:
To me it's not obvious what `arr` contains, but that may be because I'm not very familiar with QEI. However, since it's a public method, it probably wouldn't hurt to add a documentation string to this function with a quick explanation.